### PR TITLE
#15 - GetStorageSlice migrated

### LINKED ISSUE #15: Transfer GetStorageSlice
Let's move `GetStorageSlice` here from artipie/artipie.

### PR DESCRIPTION
**Pull request by olenagerasimova**
Closes #15 
Migrated `GetStorageSlice` from `artipie`.

### CODE REVIEW

File: src/main/java/com/artipie/management/RepoStorage.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+import com.artipie.asto.Storage;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Repository storage.
+ * @since 0.3
+ */
+public interface RepoStorage {
```
Comment (olegmoz) [inline src/main/java/com/artipie/management/RepoStorage.java:34]: @olenagerasimova `RepoStorage` name does not look ideal to me. Repository storage is something that is returned by this object's method `repoStorage`. Maybe naming it `Storages` or `Repositories` will be better?

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see one comment regarding new interface name

**Comment by olenagerasimova** [comment]
@olegmoz thanks, renamed

File: src/main/java/com/artipie/management/Storages.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * SOFTWARE.
+ */
+package com.artipie.management;
+
+import com.artipie.asto.Storage;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Repository storage.
```
Comment (olegmoz) [inline src/main/java/com/artipie/management/Storages.java:31]: @olenagerasimova please update JavaDoc according to new interface name

File: src/main/java/com/artipie/management/api/artifactory/GetStorageSlice.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+/**
+ * Get storage slice. See https://github.com/artipie/artipie/issues/545
+ *
+ * @since 0.3
+ */
+public final class GetStorageSlice implements Slice {
+
+    /**
+     * Artipie settings storage.
```
Comment (olegmoz) [inline src/main/java/com/artipie/management/api/artifactory/GetStorageSlice.java:51]: @olenagerasimova please update JavaDoc for the field

File: src/main/java/com/artipie/management/api/artifactory/GetStorageSlice.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+    private final Storages storage;
+
+    /**
+     * Artipie path pattern.
+     */
+    private final Pattern path;
+
+    /**
+     * New storage list slice.
+     * @param storage Artipie settings storage
```
Comment (olegmoz) [inline src/main/java/com/artipie/management/api/artifactory/GetStorageSlice.java:62]: @olenagerasimova please update JavaDoc for the parameter

**Review by olegmoz — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, please see the comments

**Comment by olenagerasimova** [comment]
@olegmoz thanks, corrected

**Review by olegmoz — APPROVED** [review]
@olenagerasimova thanks, merging

**Pull request merged**